python/testSrc/com/jetbrains/python/PyTypeTest.java

```diff
@@ -816,6 +816,12 @@ public void testDefaultTupleParameterMember() {
            "  expr, foo = xs\n");
   }
 
+  // PY-19826
+  public void testListFromTupleType() {
+    doTest("List[Union[str, int]]",
+           "t = ('1',2,3); expr = list(t)");
+  }
+
```

Comment:
Implementation of PyTypeChecker was updated so that case for expected collection and actual tuple was introduced